server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -2190,26 +2190,26 @@ private void abstractDataStoreClustersList(List<StoragePool> storagePools, List<
         }
         if (freeAddrIds.size() > 0) {
             final SearchBuilder<IPAddressVO> sb2 = _publicIpAddressDao.createSearchBuilder();
-            buildParameters(sb2, cmd);
+            buildParameters(sb2, cmd, false);
             sb2.and("ids", sb2.entity().getId(), SearchCriteria.Op.IN);
 
             SearchCriteria<IPAddressVO> sc2 = sb2.create();
             setParameters(sc2, cmd, vlanType);
             sc2.setParameters("ids", freeAddrIds.toArray());
             addrs.addAll(_publicIpAddressDao.search(sc2, searchFilter)); // Allocated + Free
         }
+        Collections.sort(addrs, Comparator.comparing(IPAddressVO::getAddress));
```

Comment:
@nvazquez @rhtyd 
do we need to order public ip address ?

Reply:
Yes it would be good

Reply:
+1